### PR TITLE
Made Additional Changes Based on Feedback for Copy Assignment Feature

### PR DESCRIPTION
**Pull request by anoverga17**
## Motivation and Context

This pull request aims to make improvements for the maintainability and style of the copy-assignments feature that was specified in the comments of pull request #5616. 

## Your Changes

**Description**:
- See feedback by david-yz-liu for pull request #5616.

**Type of change** (select all that apply):

- [x] Refactoring (internal change to codebase, without changing functionality)

## Testing

Manual testing was done to ensure the expected assignment data was being downloaded and uploaded. The data was checked after downloading by inspecting the download assignment contents and was checked after uploading by inspecting the Markus interface and database to ensure the correct settings were properly configured. 
In addition, rspec tests were run (without much change from the previous known working iteration of the copy assignments feature) to further ensure the appropriate important data was being transferred and no information was overlooked. 

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have fixed any Hound bot comments. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

File: app/models/tag.rb [hunk 1]
```diff
@@ -32,14 +32,20 @@ def self.from_csv(data, assignment_id)
...
 
-  def self.from_yml(data, assignment_id)
+  def self.from_yml(data, assignment_id, allow_ta_upload = false)
     admins = Hash[Admin.pluck(:user_name, :id)]
     begin
       tag_data = data.map do |row|
         row = row.symbolize_keys
-        name, description, user_id = row[:name], row[:description], admins[row[:user]]
+        author = admins[row[:user]]
+        # Allow TAs with proper permissions to upload yml tag data
```
Comment (david-yz-liu) [inline app/models/tag.rb:41]: I see what you're doing here, but currently we don't allow TAs to create new tags (in our `TagPolicy` the permissions require being an instructor). While it's possible to change this to allow TAs to create tags (say, if the `manage_assessments` permission is enabled), it shouldn't happen in this PR. So for now, please revert these changes to this method, including the additional parameter.

File: app/controllers/assignments_controller.rb [hunk 2]
```diff
@@ -658,17 +654,17 @@ def upload_config_files
           child_assignment.save!
           zipfile.remove(child_prop_file)
```
[the remark this answers is not in the document]
Comment (anoverga17) [reply app/controllers/assignments_controller.rb:655]: Every time I think I removed these calls they keep coming up. Thanks for catching this one.

**Review by david-yz-liu — APPROVED** [review]
